code/game/objects/items.dm

```diff
@@ -747,42 +747,9 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 
 	playsound(user.loc, 'sound/effects/extinguish.ogg', 52, 1, 7)
 
-	var/direction = get_dir(user,target)
+	SEND_SIGNAL(user, COMSIG_MOB_EXTINGUISHER_USE, target, 4)
```

Comment:
Put this before the sound and feedback, a couple of lines up.

Reply:
Same place as the movement code was before.

Reply:
This isnt movement code though, the signal should be as early in the proc as reasonable by default